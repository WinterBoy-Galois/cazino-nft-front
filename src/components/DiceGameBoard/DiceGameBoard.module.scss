@import '../../styles/shared';

.container {
  position: relative;

  > * {
    user-select: none;
  }
}

.slider {
  height: 72.6%;
  position: absolute;
  top: 7.8%;
  right: 0;
  width: 20.7%;
}
