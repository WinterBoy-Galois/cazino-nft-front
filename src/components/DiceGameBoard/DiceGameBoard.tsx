import React from 'react';
import styles from './DiceGameBoard.module.scss';
import DiceResultScale from '../icons/DiceResultScale';
import Slider from '../Slider';
import clsx from 'clsx';
import { DiceGameState } from '../../models/diceGameState.model';
import Character from './components/Character';

interface IProps {
  className?: string;
  result?: number;
  target?: number;
  disabled?: boolean;
  over?: boolean;
  onChangeTarget?: (target: number) => void;
  maxValue?: number;
  minValue?: number;
  gameState?: DiceGameState;
}

const DiceGameBoard: React.FC<IProps> = ({
  className,
  result = 0,
  target = 50,
  disabled = false,
  over = false,
  onChangeTarget,
  maxValue,
  minValue,
  gameState,
}) => {
  return (
    <div className={clsx(styles.container, className)}>
      <DiceResultScale result={result} />
      <div className={styles.slider}>
        <Slider
          disabled={disabled}
          value={target}
          switchColors={over}
          onChange={onChangeTarget}
          onUpdate={onChangeTarget}
          maxValue={maxValue}
          minValue={minValue}
        />
      </div>
      <Character gameState={gameState ?? DiceGameState.IDLE} />
    </div>
  );
};

export default DiceGameBoard;
