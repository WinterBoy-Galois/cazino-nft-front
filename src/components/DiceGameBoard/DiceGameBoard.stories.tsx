import React from 'react';
import { storiesOf } from '@storybook/react';
import DiceGameBoard from '.';
import { withKnobs, number, boolean, select } from '@storybook/addon-knobs';
import { DiceGameState } from '../../models/diceGameState.model';

storiesOf('Components/DiceGameBoard', module)
  .addDecorator(withKnobs)
  .addDecorator(storyFn => <div style={{ padding: '1rem 2rem' }}>{storyFn()}</div>)
  .add('default', () => (
    <DiceGameBoard
      result={number('Result', 55)}
      target={number('Target', 65)}
      disabled={boolean('Disabled', false)}
      over={boolean('Over', false)}
      gameState={select(
        'GameState',
        {
          IDLE: DiceGameState.IDLE,
          WON: DiceGameState.WON,
          LOST: DiceGameState.LOST,
          HITTING: DiceGameState.HITTING,
        },
        DiceGameState.IDLE
      )}
    />
  ));
