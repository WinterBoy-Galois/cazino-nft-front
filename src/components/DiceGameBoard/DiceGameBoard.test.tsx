import React from 'react';
import { render } from '@testing-library/react';
import DiceGameBoard from '.';
import { DiceGameState } from '../../models/diceGameState.model';

describe('DiceGameBoard', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <DiceGameBoard result={55} target={65} gameState={DiceGameState.WON} />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
