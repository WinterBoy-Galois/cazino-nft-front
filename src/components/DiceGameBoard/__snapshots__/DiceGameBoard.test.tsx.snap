// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiceGameBoard should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <svg
          style="width: 100%; height: 100%;"
          viewBox="0 0 616.94 669.15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
            id="Layer_2"
          >
            <g
              data-name="user scale"
              id="user_scale"
            >
              <path
                class="cls-1"
                d="M270.52 35.02H489.8v24.02H270.52zM270.52 68.77H489.8v12.01H270.52z"
              />
              <path
                class="cls-1"
                d="M289.18 31.4h137.9v51.77h-137.9z"
              />
              <path
                d="M411.92 22.72l16.81 537.36h61.07V12.52l-77.88 10.2z"
                fill="#2e475e"
              />
              <path
                d="M616.94 614.58H244.82l3.36 34.76h368.76v-34.76z"
                fill="#f39200"
              />
              <path
                class="cls-4"
                d="M416.13 560.08h200.81v54.5h-199l-1.81-54.5z"
              />
              <path
                d="M616.84 669.15H236.98l11.2-19.81h368.66v19.81z"
                fill="#b66e00"
              />
              <path
                class="cls-4"
                d="M260.87 606.79h84.82v7.79h-84.82z"
              />
              <path
                d="M268.67 580.53h69.2v26.26h-69.2z"
                fill="#be1622"
              />
              <path
                d="M313.5 580.53h13.91v26.26H313.5z"
                fill="#cb454e"
              />
              <path
                class="cls-8"
                d="M261.39 210.83L22.75 190.15a4.88 4.88 0 01-4.43-4.3L0 28.69a4.9 4.9 0 014.43-5.43L268 0a4.88 4.88 0 015.31 5l-6.61 201.08a4.88 4.88 0 01-5.31 4.75z"
              />
              <path
                d="M22.91 185.56L4.56 27.86 268.72 4.57l-6.63 201.71-239.18-20.72z"
                fill="#3e5a80"
              />
              <path
                class="cls-8"
                d="M244.62 172.16l-208.24-4.27L23.4 56.33l226.48-30.24-5.26 146.07z"
              />
              <path
                d="M40.48 163.38L28.49 60.29l216.6-28.93-4.9 136.11-199.71-4.09z"
                fill="#0d1b31"
              />
              <text
                class="text"
                x="10%"
                y="19%"
              >
                45.66
              </text>
              <path
                class="cls-1"
                d="M289.18 26.09h23.32v66.43h-23.32z"
              />
              <path
                class="cls-11"
                d="M487.91 537.78h-42.76M487.91 528.07h-13.96M487.91 518.37h-13.96M487.91 508.67h-13.96M487.91 498.97h-13.96M487.91 489.27h-42.76M487.91 479.57h-13.96M487.91 469.86h-13.96M487.91 460.16h-13.96M487.91 450.46h-13.96M487.91 440.76h-42.76M487.91 431.06h-13.96M487.91 421.36h-13.96M487.91 411.65h-13.96M487.91 401.95h-13.96M487.91 392.25h-42.76M487.91 382.55h-13.96M487.91 372.85h-13.96M487.91 363.15h-13.96M487.91 353.44h-13.96M487.91 343.74h-42.76M487.91 334.04h-13.96M487.91 324.34h-13.96M487.91 314.64h-13.96M487.91 304.94h-13.96M487.91 295.23h-42.76M487.91 285.53h-13.96M487.91 275.83h-13.96M487.91 266.13h-13.96M487.91 256.43h-13.96M487.91 246.73h-42.76M487.91 237.02h-13.96M487.91 227.32h-13.96M487.91 217.62h-13.96M487.91 207.92h-13.96M487.91 198.22h-42.76M487.91 188.51h-13.96M487.91 178.81h-13.96M487.91 169.11h-13.96M487.91 159.41h-13.96M487.91 149.71h-42.76M487.91 140.01h-13.96M487.91 130.3h-13.96M487.91 120.6h-13.96M487.91 110.9h-13.96M487.91 101.2h-42.76M487.91 91.5h-13.96M487.91 81.8h-13.96M487.91 72.09h-13.96M487.91 62.39h-13.96M487.91 52.69h-42.76"
              />
              <path
                class="cls-12"
                d="M445.74 31.64h2.39a.29.29 0 01.22.09.25.25 0 01.08.21v16.9a.28.28 0 01-.08.21.29.29 0 01-.22.09h-2.29a.29.29 0 01-.22-.09.28.28 0 01-.08-.21v-14.3s0-.06-.05-.09a.11.11 0 00-.11 0l-1.77.52a.21.21 0 01-.12 0c-.15 0-.23-.09-.23-.28v-1.63a.37.37 0 01.23-.35l1.87-1a1.06 1.06 0 01.38-.07zM451.43 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.29 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM462.18 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.26 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM458.76 286.24a12.43 12.43 0 01-.1 1.75 3.84 3.84 0 01-1.3 2.85 4.62 4.62 0 01-3.12 1 4.5 4.5 0 01-3-.95 4.19 4.19 0 01-1.4-2.85l-.08-.7a.27.27 0 01.3-.3h2.3a.3.3 0 01.33.3 2.32 2.32 0 00.05.45c.2 1 .7 1.55 1.5 1.55a1.34 1.34 0 001-.4 2 2 0 00.51-1.12 10.08 10.08 0 00.1-1.6 7.44 7.44 0 00-.15-1.6 1.8 1.8 0 00-.51-1 1.36 1.36 0 00-1-.36 1.49 1.49 0 00-1.55 1.15.32.32 0 01-.33.28H450a.26.26 0 01-.21-.09.29.29 0 01-.09-.21v-9.93a.29.29 0 01.09-.21.26.26 0 01.21-.09h8.15a.31.31 0 01.3.3v1.9a.26.26 0 01-.09.21.29.29 0 01-.21.09h-5.43c-.08 0-.12 0-.12.13v4.87c0 .05 0 .08.05.09s.06 0 .1 0a3 3 0 011-.71 3.09 3.09 0 011.28-.26 3.19 3.19 0 012.42 1 4.48 4.48 0 011.08 2.71 10 10 0 01.23 1.75zM461.44 290.58a4.87 4.87 0 01-1.28-3.51v-8.35a4.91 4.91 0 011.28-3.52 4.44 4.44 0 013.37-1.33 4.53 4.53 0 013.42 1.33 4.91 4.91 0 011.28 3.52v8.35a4.87 4.87 0 01-1.28 3.51 4.54 4.54 0 01-3.42 1.34 4.44 4.44 0 01-3.37-1.34zm4.69-1.78a2.38 2.38 0 00.48-1.56v-8.7a2.38 2.38 0 00-.48-1.56 1.65 1.65 0 00-1.32-.59 1.55 1.55 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.52 1.52 0 001.27.59 1.62 1.62 0 001.32-.59zM461.09 532.63a4.87 4.87 0 01-1.27-3.51v-8.35a4.87 4.87 0 011.27-3.51 4.47 4.47 0 013.38-1.34 4.52 4.52 0 013.41 1.34 4.84 4.84 0 011.29 3.51v8.35a4.84 4.84 0 01-1.29 3.51 4.48 4.48 0 01-3.41 1.34 4.43 4.43 0 01-3.38-1.34zm4.69-1.77a2.33 2.33 0 00.49-1.56v-8.7a2.38 2.38 0 00-.49-1.57 1.6 1.6 0 00-1.31-.58 1.57 1.57 0 00-1.28.58 2.49 2.49 0 00-.47 1.57v8.7a2.44 2.44 0 00.47 1.56 1.58 1.58 0 001.28.59 1.61 1.61 0 001.31-.59z"
              />
            </g>
          </g>
        </svg>
        <div
          class="slider"
        >
          <div
            class="slider__wrapper"
          >
            <div
              class="slider"
            >
              <div
                class="slider__rail"
              />
              <div
                class="slider-handles"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="50"
                  class="handle"
                  role="slider"
                  style="top: 50%;"
                >
                  <svg
                    class="handle__small"
                    height="41.562"
                    width="88.143"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26.451 41.547h-.064V41.5L0 20.774 26.387.053V0H80.142a8 8 0 018 8v25.547a8 8 0 01-8 8H26.471v.015z"
                      fill="#2a434f"
                    />
                    <text
                      class="handle__value"
                      x="24%"
                      y="70%"
                    >
                      50.00
                    </text>
                  </svg>
                </div>
              </div>
              <div
                class="slider-tracks"
              >
                <div
                  class="track"
                  style="top: 0%; height: 50%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <svg
        style="width: 100%; height: 100%;"
        viewBox="0 0 616.94 669.15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
          id="Layer_2"
        >
          <g
            data-name="user scale"
            id="user_scale"
          >
            <path
              class="cls-1"
              d="M270.52 35.02H489.8v24.02H270.52zM270.52 68.77H489.8v12.01H270.52z"
            />
            <path
              class="cls-1"
              d="M289.18 31.4h137.9v51.77h-137.9z"
            />
            <path
              d="M411.92 22.72l16.81 537.36h61.07V12.52l-77.88 10.2z"
              fill="#2e475e"
            />
            <path
              d="M616.94 614.58H244.82l3.36 34.76h368.76v-34.76z"
              fill="#f39200"
            />
            <path
              class="cls-4"
              d="M416.13 560.08h200.81v54.5h-199l-1.81-54.5z"
            />
            <path
              d="M616.84 669.15H236.98l11.2-19.81h368.66v19.81z"
              fill="#b66e00"
            />
            <path
              class="cls-4"
              d="M260.87 606.79h84.82v7.79h-84.82z"
            />
            <path
              d="M268.67 580.53h69.2v26.26h-69.2z"
              fill="#be1622"
            />
            <path
              d="M313.5 580.53h13.91v26.26H313.5z"
              fill="#cb454e"
            />
            <path
              class="cls-8"
              d="M261.39 210.83L22.75 190.15a4.88 4.88 0 01-4.43-4.3L0 28.69a4.9 4.9 0 014.43-5.43L268 0a4.88 4.88 0 015.31 5l-6.61 201.08a4.88 4.88 0 01-5.31 4.75z"
            />
            <path
              d="M22.91 185.56L4.56 27.86 268.72 4.57l-6.63 201.71-239.18-20.72z"
              fill="#3e5a80"
            />
            <path
              class="cls-8"
              d="M244.62 172.16l-208.24-4.27L23.4 56.33l226.48-30.24-5.26 146.07z"
            />
            <path
              d="M40.48 163.38L28.49 60.29l216.6-28.93-4.9 136.11-199.71-4.09z"
              fill="#0d1b31"
            />
            <text
              class="text"
              x="10%"
              y="19%"
            >
              45.66
            </text>
            <path
              class="cls-1"
              d="M289.18 26.09h23.32v66.43h-23.32z"
            />
            <path
              class="cls-11"
              d="M487.91 537.78h-42.76M487.91 528.07h-13.96M487.91 518.37h-13.96M487.91 508.67h-13.96M487.91 498.97h-13.96M487.91 489.27h-42.76M487.91 479.57h-13.96M487.91 469.86h-13.96M487.91 460.16h-13.96M487.91 450.46h-13.96M487.91 440.76h-42.76M487.91 431.06h-13.96M487.91 421.36h-13.96M487.91 411.65h-13.96M487.91 401.95h-13.96M487.91 392.25h-42.76M487.91 382.55h-13.96M487.91 372.85h-13.96M487.91 363.15h-13.96M487.91 353.44h-13.96M487.91 343.74h-42.76M487.91 334.04h-13.96M487.91 324.34h-13.96M487.91 314.64h-13.96M487.91 304.94h-13.96M487.91 295.23h-42.76M487.91 285.53h-13.96M487.91 275.83h-13.96M487.91 266.13h-13.96M487.91 256.43h-13.96M487.91 246.73h-42.76M487.91 237.02h-13.96M487.91 227.32h-13.96M487.91 217.62h-13.96M487.91 207.92h-13.96M487.91 198.22h-42.76M487.91 188.51h-13.96M487.91 178.81h-13.96M487.91 169.11h-13.96M487.91 159.41h-13.96M487.91 149.71h-42.76M487.91 140.01h-13.96M487.91 130.3h-13.96M487.91 120.6h-13.96M487.91 110.9h-13.96M487.91 101.2h-42.76M487.91 91.5h-13.96M487.91 81.8h-13.96M487.91 72.09h-13.96M487.91 62.39h-13.96M487.91 52.69h-42.76"
            />
            <path
              class="cls-12"
              d="M445.74 31.64h2.39a.29.29 0 01.22.09.25.25 0 01.08.21v16.9a.28.28 0 01-.08.21.29.29 0 01-.22.09h-2.29a.29.29 0 01-.22-.09.28.28 0 01-.08-.21v-14.3s0-.06-.05-.09a.11.11 0 00-.11 0l-1.77.52a.21.21 0 01-.12 0c-.15 0-.23-.09-.23-.28v-1.63a.37.37 0 01.23-.35l1.87-1a1.06 1.06 0 01.38-.07zM451.43 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.29 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM462.18 48.08a5 5 0 01-1.27-3.52v-8.35a4.91 4.91 0 011.27-3.51 5 5 0 016.79 0 4.84 4.84 0 011.29 3.51v8.35a4.88 4.88 0 01-1.26 3.52 5 5 0 01-6.79 0zm4.69-1.78a2.33 2.33 0 00.49-1.56V36a2.33 2.33 0 00-.49-1.56 1.58 1.58 0 00-1.31-.59 1.52 1.52 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.55 1.55 0 001.27.59 1.61 1.61 0 001.31-.55zM458.76 286.24a12.43 12.43 0 01-.1 1.75 3.84 3.84 0 01-1.3 2.85 4.62 4.62 0 01-3.12 1 4.5 4.5 0 01-3-.95 4.19 4.19 0 01-1.4-2.85l-.08-.7a.27.27 0 01.3-.3h2.3a.3.3 0 01.33.3 2.32 2.32 0 00.05.45c.2 1 .7 1.55 1.5 1.55a1.34 1.34 0 001-.4 2 2 0 00.51-1.12 10.08 10.08 0 00.1-1.6 7.44 7.44 0 00-.15-1.6 1.8 1.8 0 00-.51-1 1.36 1.36 0 00-1-.36 1.49 1.49 0 00-1.55 1.15.32.32 0 01-.33.28H450a.26.26 0 01-.21-.09.29.29 0 01-.09-.21v-9.93a.29.29 0 01.09-.21.26.26 0 01.21-.09h8.15a.31.31 0 01.3.3v1.9a.26.26 0 01-.09.21.29.29 0 01-.21.09h-5.43c-.08 0-.12 0-.12.13v4.87c0 .05 0 .08.05.09s.06 0 .1 0a3 3 0 011-.71 3.09 3.09 0 011.28-.26 3.19 3.19 0 012.42 1 4.48 4.48 0 011.08 2.71 10 10 0 01.23 1.75zM461.44 290.58a4.87 4.87 0 01-1.28-3.51v-8.35a4.91 4.91 0 011.28-3.52 4.44 4.44 0 013.37-1.33 4.53 4.53 0 013.42 1.33 4.91 4.91 0 011.28 3.52v8.35a4.87 4.87 0 01-1.28 3.51 4.54 4.54 0 01-3.42 1.34 4.44 4.44 0 01-3.37-1.34zm4.69-1.78a2.38 2.38 0 00.48-1.56v-8.7a2.38 2.38 0 00-.48-1.56 1.65 1.65 0 00-1.32-.59 1.55 1.55 0 00-1.27.59 2.38 2.38 0 00-.48 1.56v8.7a2.38 2.38 0 00.48 1.56 1.52 1.52 0 001.27.59 1.62 1.62 0 001.32-.59zM461.09 532.63a4.87 4.87 0 01-1.27-3.51v-8.35a4.87 4.87 0 011.27-3.51 4.47 4.47 0 013.38-1.34 4.52 4.52 0 013.41 1.34 4.84 4.84 0 011.29 3.51v8.35a4.84 4.84 0 01-1.29 3.51 4.48 4.48 0 01-3.41 1.34 4.43 4.43 0 01-3.38-1.34zm4.69-1.77a2.33 2.33 0 00.49-1.56v-8.7a2.38 2.38 0 00-.49-1.57 1.6 1.6 0 00-1.31-.58 1.57 1.57 0 00-1.28.58 2.49 2.49 0 00-.47 1.57v8.7a2.44 2.44 0 00.47 1.56 1.58 1.58 0 001.28.59 1.61 1.61 0 001.31-.59z"
            />
          </g>
        </g>
      </svg>
      <div
        class="slider"
      >
        <div
          class="slider__wrapper"
        >
          <div
            class="slider"
          >
            <div
              class="slider__rail"
            />
            <div
              class="slider-handles"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="50"
                class="handle"
                role="slider"
                style="top: 50%;"
              >
                <svg
                  class="handle__small"
                  height="41.562"
                  width="88.143"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M26.451 41.547h-.064V41.5L0 20.774 26.387.053V0H80.142a8 8 0 018 8v25.547a8 8 0 01-8 8H26.471v.015z"
                    fill="#2a434f"
                  />
                  <text
                    class="handle__value"
                    x="24%"
                    y="70%"
                  >
                    50.00
                  </text>
                </svg>
              </div>
            </div>
            <div
              class="slider-tracks"
            >
              <div
                class="track"
                style="top: 0%; height: 50%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
